src/index.js

```diff
@@ -6,11 +6,9 @@ import { BrowserRouter } from 'react-router-dom';
 
 const root = ReactDOM.createRoot(document.getElementById('root'));
 root.render(
-  <React.StrictMode>
-    <BrowserRouter>
-      <App />
-    </BrowserRouter>
-  </React.StrictMode>,
```

Comment:
@behrouz2115 This is what was causing useEffect to run twice in React 18. So you don't need to do `ref.hasRun` anymore. I was wrong 🙏

Reply:
thank you James it was good point.